### PR TITLE
feat: Make natural path structure

### PR DESCRIPTION
**Pull request by goisaki**
## changes
  - Move all documents except `snippets/` to `docs/`
  - Change NotFound to Layout to inject redirect processing
  - Inject redirect processing into Layout component

### CODE REVIEW

**Comment by goisaki** [comment]
@yusukebe, Could you create the `next` branch for this and related future PRs?

**Comment by yusukebe** [comment]
@goisaki 

Okay, just a moment!

**Comment by goisaki** [comment]
@yusukebe 
Resolved conflict to main branch!

**Comment by yusukebe** [comment]
Hi @goisaki 

Sorry for the late. I'm considering the documentation for the Hono recently.

I'd like to discuss API documents before merging this PR. I noticed that the documentation generated by TSDoc/JSDoc would be good enough for the API (only API). So, I think if we have the docs by TSDoc/JSDoc, we don't need the API section for the doc. How do you think about it? Anyway, we have to discuss it elsewhere.

**Comment by goisaki** [comment]
I see. Let's continue with the discussion.

**Comment by goisaki** [comment]
ref: https://github.com/orgs/honojs/discussions/2854

**Comment by yusukebe** [comment]
Hey @goisaki 

Let's restart to work on this matter. I've created the `next` branch. Can you resolve the conflicts? Then, I'll merge this into the `next` and will work on other ideas that I'm thinking of now.

File: .vitepress/theme/Layout.vue [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+const { page } = useData()
+const router = useRouter()
+
+const redirects: {
+  pattern: RegExp
+  to: string
+}[] = [
+  {
+    pattern: /^(?=\/(api|concepts|getting-started|guides|helpers|middleware))/,
+    to: '/docs/$1',
```
[the remark this answers is not in the document]
Comment (goisaki) [reply .vitepress/theme/Layout.vue]: Oh, thanks for letting me know! Thanks for letting me know, I fixed it at `2effa8e7`.

**Comment by yusukebe** [comment]
Thanks @goisaki !

Regarding redirecting, it's better to use Cloudflare Pages's redirect feature, which returns a Response with a `301` code.

https://developers.cloudflare.com/pages/configuration/redirects/

**Comment by NicoPlyley** [comment]
> it's better to use Cloudflare Pages's redirect feature

Can you confirm it is working properly. I have tired to use it a few times, it says they were loaded in but it did not work. It's been about 1 year since I've last tired. I've resulting in just using page rules or bulk redirects instead

**Comment by yusukebe** [comment]
Sorry, I accidentally closed it. Reopened.

**Comment by yusukebe** [comment]
@NicoPlyley @goisaki 

`_redirects` works well!

* The change: https://github.com/honojs/website/pull/389/files
* Demo: <https://bef8a4e3.hono.pages.dev/home301>

**Comment by goisaki** [comment]
Hi @yusukebe @NicoPlyley
I made the change in https://github.com/honojs/website/commit/c71f9cb191b5b7d789804ba2542878cf3f45a100.
Currently this PR includes the deletion of `.vitepress/theme/NotFound.vue`, is there any problem?

**Comment by yusukebe** [comment]
@goisaki Thanks

> Currently this PR includes the deletion of `.vitepress/theme/NotFound.vue`, is there any problem?

I think we don't need to remove it. Do you have any reason to delete it?

**Comment by goisaki** [comment]
hi @yusukebe 
The display looks exactly the same as the default NotFound page. So I saw no positive reason to keep this file. Is there an intent to keep it?

**Comment by yusukebe** [comment]
> The display looks exactly the same as the default NotFound page. So I saw no positive reason to keep this file. Is there an intent to keep it?

Ahh, deleting is okay. It was necessary for the old version of VitePress. Thanks!

**Comment by goisaki** [comment]
Thanks for agreeing. Ready to go!

File: public/_redirects [hunk 2]
```diff
@@ -1,2 +1,9 @@
...
\ No newline at end of file
+/top / 301
+/snippets/top 301
+
+/api/* /:splat 301
+/concepts/* /:splat 301
+/getting-started/* /:splat 301
+/guides/* /:splat 301
+/helpers/* /:splat 301
+/middleware/* /:splat 301
```
Comment (NicoPlyley) [inline public/_redirects:9]: Testing this wrangler does not work. When I go to `/guides/jsx` it redirects me to `/jsx`

The following is working for me:
```
/api/* /docs/api/:splat 301
/concepts/* /docs/concepts/:splat 301
/getting-started/* /docs/getting-started/:splat 301
/guides/* /docs/guides/:splat 301
/helpers/* /docs/helpers/:splat 301
/middleware/* /docs/middleware/:splat 301
```

Comment (goisaki) [reply public/_redirects]: Sorry for the delay, fixed in https://github.com/honojs/website/commit/194e9d8a89c439996ddcd5ea724959281dd041bf.

**Comment by yusukebe** [comment]
@goisaki 

Thanks! I'll merge this now into the `next`.

**Pull request merged**